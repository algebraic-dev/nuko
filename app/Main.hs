module Main (main) where

import Relude (IO, undefined)

main :: IO ()
main = undefined