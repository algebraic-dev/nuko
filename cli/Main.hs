module Main (main) where

import Relude
import Nuko.Driver            (compile, mainMod)
import Data.Aeson.Text        (encodeToLazyText)
import Data.Aeson             (ToJSON)
import Data.These             (These(These, That, This) )
import Nuko.Report.Render     (renderDiagnostic)

main :: MonadIO m => m ()
main = do
  content <- readFileBS "./examples/normal.nk"
  let res    = compile mainMod "examples/normal.nk" content
  let source = lines (decodeUtf8 content)
  case res of
    That _    -> putTextLn   "✓"
    These e _ -> putTextLn $ "•\n"  <> unlines (map (renderDiagnostic source) e)
    This e    -> putTextLn $ "✗\n"  <> unlines (map (renderDiagnostic source) e)