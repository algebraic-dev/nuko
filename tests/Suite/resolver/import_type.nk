type A =
  | B
  | C

import Main.A (B)