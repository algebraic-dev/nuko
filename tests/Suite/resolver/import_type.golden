✓ That
 └Prod
   ├Program
   | ├typeDecls:
   | | └List
   | |   └TypeDecl
   | |     ├tyName: (Name: TyName A 0:5-0:6)
   | |     ├tyArgs: Empty List
   | |     └tyDecl:
   | |       └TypeSum
   | |         └NonEmpty
   | |           ├Prod
   | |           | ├Name: ConsName B 1:4-1:5
   | |           | └Empty List
   | |           └Prod
   | |             ├Name: ConsName C 2:4-2:5
   | |             └Empty List
   | ├letDecls: Empty List
   | ├impDecls: Empty List
   | └programExt: NoExt
   └ResolverState
     ├_openedNames:
     | └HashMap
     |   ├Prod
     |   | ├Name: TyName Int 0:0-0:0
     |   | └Single
     |   |   └Qualified: Prelude 0:0-0:0
     |   |     └Name: TyName Int 0:0-0:0
     |   ├Prod
     |   | ├Name: ConsName B 4:15-4:16
     |   | └Single
     |   |   └Qualified: Main.A 4:15-4:16
     |   |     └Name: ConsName B 4:15-4:16
     |   └Prod
     |     ├Name: TyName String 0:0-0:0
     |     └Single
     |       └Qualified: Prelude 0:0-0:0
     |         └Name: TyName String 0:0-0:0
     ├_openedModules:
     | └HashMap
     |   ├Prod
     |   | ├ModName: A
     |   | └NameSpace
     |   |   ├_modName: (ModName: Main.A)
     |   |   └_names:
     |   |     └HashMap
     |   |       ├Prod
     |   |       | ├Name: ConsName C 2:4-2:5
     |   |       | └Public
     |   |       └Prod
     |   |         ├Name: ConsName B 1:4-1:5
     |   |         └Public
     |   └Prod
     |     ├ModName: Main.A
     |     └NameSpace
     |       ├_modName: (ModName: Main.A)
     |       └_names:
     |         └HashMap
     |           ├Prod
     |           | ├Name: ConsName C 2:4-2:5
     |           | └Public
     |           └Prod
     |             ├Name: ConsName B 1:4-1:5
     |             └Public
     ├_localNames: HashMap
     ├_usedLocals: HashMap
     ├_currentNamespace:
     | └NameSpace
     |   ├_modName: (ModName: Main)
     |   └_names:
     |     └HashMap
     |       └Prod
     |         ├Name: TyName A 0:5-0:6
     |         └Public
     └_newNamespaces:
       └HashMap
         └Prod
           ├ModName: Main.A
           └NameSpace
             ├_modName: (ModName: Main.A)
             └_names:
               └HashMap
                 ├Prod
                 | ├Name: ConsName C 2:4-2:5
                 | └Public
                 └Prod
                   ├Name: ConsName B 1:4-1:5
                   └Public
